test-framework/sudo-compliance-tests/src/sudoers/secure_path.rs

```diff
@@ -0,0 +1,91 @@
+use sudo_test::{Command, Env, TextFile};
+
+use crate::{Result, SUDOERS_ALL_ALL_NOPASSWD};
+
+#[test]
+fn if_unset_searches_program_in_invoking_users_path() -> Result<()> {
+    let path = "/root/my-script";
+    let env = Env(SUDOERS_ALL_ALL_NOPASSWD)
+        .file(path, TextFile("#!/bin/sh").chmod("100"))
+        .build()?;
+
+    Command::new("sh")
+        .args(["-c", "export PATH=/root; cd /; /usr/bin/sudo my-script"])
+        .exec(&env)?
+        .assert_success()?;
+
+    Ok(())
+}
+
+#[test]
+fn if_set_searches_program_in_secure_path() -> Result<()> {
+    let path = "/root/my-script";
+    let env = Env("\
+Defaults secure_path=.:/root
+ALL ALL=(ALL:ALL) NOPASSWD: ALL")
+    .file(path, TextFile("#!/bin/sh").chmod("100"))
+    .build()?;
+
+    let relative_path = "unset PATH; cd /bin; /usr/bin/sudo true";
```

Comment:
Shouldn't we also have a test where we do have  PATH set, to see that it doesn't override the secure_path?